### PR TITLE
Modal service improvements

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**
Pass modal props in `showModal` function

### CODE REVIEW

File: universal-login-react/src/App.tsx [hunk 1]
```diff
@@ -77,8 +78,8 @@ export const App = () => {
               render={() => (
                 <>
                   <ReactModalContext.Provider value={modalService}>
-                    <button onClick={() => modalService.showModal('topUpAccount')}>Show Topup</button>
-                    <Modal contractAddress={Wallet.createRandom().address}/>
+                    <button onClick={() => modalService.showModal('topUpAccount', {contractAddress: Wallet.createRandom().address, onRampConfig: relayerConfig!.onRampProviders})}>Show Topup</button>
```
Comment (marekkirejczyk) [inline universal-login-react/src/App.tsx]: Shall we add:
```ts
if (!relayerConfig) {
  return '<div>Loading...</div>'
} 
```

File: universal-login-react/src/ui/Modals/Modal.tsx [hunk 2]
```diff
@@ -2,14 +2,13 @@ import React, {useContext} from 'react';
 import {TopUp} from '../TopUp/TopUp';
 import {ModalWrapper} from './ModalWrapper';
 import WaitingFor from '../commons/WaitingFor';
-import {ReactModalContext} from '../../core/models/ReactModalContext';
+import {ReactModalContext, TopUpProps} from '../../core/models/ReactModalContext';
 
 export interface ModalProps {
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Modals/Modal.tsx]: Modals

File: universal-login-react/src/ui/Modals/Modal.tsx [hunk 3]
```diff
@@ -2,14 +2,13 @@ import React, {useContext} from 'react';
...
   modalClassName?: string;
-  contractAddress?: string;
 }
 
-const Modal = ({modalClassName, contractAddress}: ModalProps) => {
+const Modal = ({modalClassName}: ModalProps) => {
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Modals/Modal.tsx]: Modals

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 4]
```diff
@@ -15,17 +15,16 @@ interface OnboardingProps {
...
   const onConnectClick = () => {
     onConnect();
   };
 
   const onCreateClick = async (ensName: string) => {
     const {deploy, waitForBalance, contractAddress} = await walletService.createFutureWallet();
-    setContractAddress(contractAddress);
-    modalService.showModal('topUpAccount');
+    const relayerConfig = await sdk.getRelayerConfig();
+    modalService.showModal('topUpAccount', {contractAddress, onRampConfig: relayerConfig!.onRampProviders});
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx]: extract `topupProps`?`

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx [hunk 5]
```diff
@@ -44,7 +43,7 @@ export const Onboarding = ({sdk, onConnect, onCreate, domains, className}: Onboa
             onConnectClick={onConnectClick}
             domains={domains}
           />
-          <Modal contractAddress={contractAddress}/>
+          <Modal />
```
Comment (marekkirejczyk) [inline universal-login-react/src/ui/Onboarding/Onboarding.tsx]: 👍

File: universal-login-wallet/src/core/entities/WalletModalContext.ts [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
-import {IModalService} from '@universal-login/react/dist/src/core/services/createModalService';
+import {IModalService} from '@universal-login/react';
 
 export type WalletModalType = 'transfer' | 'request' | 'invitation' | 'topUpAccount' | 'address' | 'personalInfo' | 'cardInfo' | 'waiting' | 'waitingForDeploy' | 'waitingForTransfer' | 'transactionSuccess' | 'safello' | 'none';
 
-export const WalletModalContext = React.createContext({} as IModalService<WalletModalType>);
+export const WalletModalContext = React.createContext({} as IModalService<WalletModalType, 'none'>);
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/core/entities/WalletModalContext.ts]: 'none' -> void

File: universal-login-wallet/src/ui/react/App.tsx [hunk 7]
```diff
@@ -15,11 +16,10 @@ import {TermsAndConditionsScreen} from './Home/TermsAndConditionsScreen';
 import {CreateAccount} from './CreateAccount/CreateAccount';
 import {ConnectAccount} from './ConnectAccount/ConnectAccount';
 import {ConnectWithEmoji} from './ConnectAccount/ConnectWithEmoji';
-import {createModalService} from '@universal-login/react';
 import {WalletModalContext, WalletModalType} from '../../core/entities/WalletModalContext';
 
 const App = () => {
-  const modalService = createModalService<WalletModalType>();
+  const modalService = createModalService<WalletModalType, 'none'>();
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/react/App.tsx]: 'none' -> void

File: universal-login-react/src/core/services/createModalService.ts [hunk 8]
```diff
@@ -1,16 +1,27 @@
 import {useState} from 'react';
 
-export interface IModalService<T> {
+export interface IModalService<T, K> {
   modalState: T | 'none';
-  showModal: (name: T | 'none') => void;
+  modalProps: K | 'none';
```
Comment (gasiortomasz) [inline universal-login-react/src/core/services/createModalService.ts]: `modalProps: L | {};` ?

File: universal-login-react/src/core/services/createModalService.ts [hunk 9]
```diff
@@ -1,16 +1,27 @@
...
+export const createModalService = <T, K>(): IModalService<T, K> => {
   const [modalState, setModalState] = useState<T | 'none'>('none');
+  const [modalProps, setModalProps] = useState<K | 'none'>('none');
+  const showModal = (modalState: T | 'none', props?: K) => {
+    props ? setModalProps(props) : setModalProps('none');
+    setModalState(modalState);
+  };
+  const hideModal = () => {
+    setModalProps('none');
+    setModalState('none');
```
Comment (gasiortomasz) [inline universal-login-react/src/core/services/createModalService.ts]: `setModalProps({});`?

File: universal-login-react/src/App.tsx (hunk 1 above)
Comment (JustynaBroniszewska) [reply universal-login-react/src/App.tsx]: ✅

File: universal-login-react/src/ui/Modals/Modal.tsx (hunk 2 above)
Comment (JustynaBroniszewska) [reply universal-login-react/src/ui/Modals/Modal.tsx]: ✅

File: universal-login-react/src/ui/Modals/Modal.tsx (hunk 3 above)
Comment (JustynaBroniszewska) [reply universal-login-react/src/ui/Modals/Modal.tsx]: ✅

File: universal-login-react/src/ui/Onboarding/Onboarding.tsx (hunk 4 above)
Comment (JustynaBroniszewska) [reply universal-login-react/src/ui/Onboarding/Onboarding.tsx]: ✅

File: universal-login-react/src/core/services/createModalService.ts (hunk 9 above)
Comment (marekkirejczyk) [reply universal-login-react/src/core/services/createModalService.ts]: Yes, let's fix it in the next PR.

**Pull request merged**